centreon/src/Core/ServiceGroup/Infrastructure/Repository/DbReadServiceGroupRepository.php

```diff
@@ -298,6 +487,318 @@ public function findByIds(int ...$serviceGroupIds): array
+            SQL;
+
+        $statement = $this->db->prepare($this->translateDbName($request));
+
+        foreach ($bindValues as $key => $value) {
+            $statement->bindValue($key, $value, \PDO::PARAM_INT);
+        }
+
+        $statement->execute();
+
+        while (false !== ($hasAccessToAll = $statement->fetchColumn())) {
+            if (true === (bool) $hasAccessToAll) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param int[] $accessGroupIds
+     *
+     * @return string
+     */
+    private function generateHostAclSubRequest(array $accessGroupIds = []): string
+    {
+        $subRequest = '';
+
+        if (
+            $accessGroupIds !== []
+            && ! $this->hasAccessToAllHosts($accessGroupIds)
+        ) {
+            [, $bindQuery] = $this->createMultipleBindQuery($accessGroupIds, ':access_group_id_');
+
+            $subRequest = <<<SQL
+                AND sgr.host_host_id IN (
+                    SELECT arhr.host_host_id
+                    FROM `:db`.acl_resources_host_relations arhr
+                    INNER JOIN `:db`.acl_resources res
+                        ON res.acl_res_id = arhr.acl_res_id
+                    INNER JOIN `:db`.acl_res_group_relations argr
+                        ON argr.acl_res_id = res.acl_res_id
+                    INNER JOIN `:db`.acl_groups ag
+                        ON ag.acl_group_id = argr.acl_group_id
+                    WHERE ag.acl_group_id IN ({$bindQuery})
+                )
+                SQL;
+        }
+
+        return $subRequest;
+    }
+
+    /**
+     * @param int[] $accessGroupIds
+     *
+     * @return string
+     */
+    private function generateServiceCategoryAclSubRequest(array $accessGroupIds = []): string
+    {
+        $request = '';
+
+        if (
+            $accessGroupIds !== []
+            && ! $this->hasAccessToAllServiceCategories($accessGroupIds)
+        ) {
+            [, $bindQuery] = $this->createMultipleBindQuery($accessGroupIds, ':access_group_id_');
+
+            $request = <<<SQL
+                    AND scr.sc_id IN (
+                        SELECT arscr.sc_id
+                        FROM `:db`.acl_resources_sc_relations arscr
+                        INNER JOIN `:db`.acl_resources res
+                            ON res.acl_res_id = arscr.acl_res_id
+                        INNER JOIN `:db`.acl_res_group_relations argr
+                            ON argr.acl_res_id = res.acl_res_id
+                        INNER JOIN `:db`.acl_groups ag
+                            ON ag.acl_group_id = argr.acl_group_id
+                        WHERE ag.acl_group_id IN ({$bindQuery})
+                    )
+                SQL;
+        }
+
+        return $request;
+    }
+
+    /**
+     * @param int[] $accessGroupIds
+     *
+     * @return string
+     */
+    private function generateHostCategoryAclSubRequest(array $accessGroupIds = []): string
```

Comment:
This method must be in the Host repository or in a Hostcategory repository trait.